buildSrc/src/main/java/org/springframework/boot/build/MavenPomConventions.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.build;
+
+import org.apache.maven.artifact.repository.MavenArtifactRepository;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.plugins.JavaPluginExtension;
+import org.gradle.api.publish.PublishingExtension;
+import org.gradle.api.publish.maven.MavenPom;
+import org.gradle.api.publish.maven.MavenPomDeveloperSpec;
+import org.gradle.api.publish.maven.MavenPomIssueManagement;
+import org.gradle.api.publish.maven.MavenPomLicenseSpec;
+import org.gradle.api.publish.maven.MavenPomOrganization;
+import org.gradle.api.publish.maven.MavenPomScm;
+import org.gradle.api.publish.maven.MavenPublication;
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin;
+
+/**
+ * Plugin to apply Maven Publishing conventions to projects that are part of Spring Boot's
+ * build. Conventions are applied in response to various plugins being applied.
+ *
+ * When the {@link MavenPublishPlugin Maven Publish plugin} is applied:
+ *
+ * <ul>
+ * <li>If the {@code deploymentRepository} property has been set, a
+ * {@link MavenArtifactRepository Maven artifact repository} is configured to publish to
+ * it.
+ * <li>The poms of all {@link MavenPublication Maven publications} are customized to meet
+ * Maven Central's requirements.
+ * <li>If the {@link JavaPlugin Java plugin} has also been applied, creation of Javadoc
+ * and source jars is enabled.
+ * </ul>
+ *
+ * <p/>
+ *
+ * @author Mike Smithson
+ */
+public class MavenPomConventions implements Plugin<Project> {
```

Comment:
This class doesn't need to be a `Plugin`. It can also be package-private. The existing `AsciidoctorConventions` class sets a good precedent here.

Reply:
This class applies more than pom conventions. I think `MavenPublishingConventions` would be a better name.